app/page/info/tokens.tsx

```diff
@@ -0,0 +1,193 @@
+import { faCaretDown, faXmark } from "@fortawesome/free-solid-svg-icons"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import * as ethers from "ethers"
+import { useState, type ChangeEvent, type FormEvent } from "react"
+
+import { useProviderContext } from "~app/context/provider"
+import { useAccount, useAction, useTokens } from "~app/storage"
+import { getChainName } from "~packages/network/util/chain"
+import type { BigNumberish, HexString } from "~typing"
+
+export function Tokens() {
+  const tokens = useTokens()
+  const account = useAccount()
+  const explorerUrl = `https://${getChainName(account.chainId)}.etherscan.io/`
+
+  const [isTokenModalOpened, setIsTokenModalOpened] = useState<boolean>(false)
+  const toggleAccountModal = () => setIsTokenModalOpened(!isTokenModalOpened)
+
+  return (
+    <div>
+      <div className="flex-col justify-center items-center h-auto p-3 border-0 rounded-lg text-base">
+        Tokens:
+        {tokens &&
+          tokens.map((token, index) => {
```

Comment:
nit: tokens from `useTokens` is already an array, so we can directly `map` on it

```suggestion
        {tokens.map((token, index) => {
```